p2p/dandelion/src/tests/router.rs

```diff
@@ -0,0 +1,204 @@
+use std::time::Duration;
+
+use tower::{Service, ServiceExt};
+
+use crate::{DandelionConfig, DandelionRouteReq, DandelionRouterBuilder, Graph, TxState};
+
+use super::*;
+
+#[tokio::test]
+async fn routes_consistent() {
+    // make sure a tx from the same peer goes to the same peer.
+
+    let config = DandelionConfig {
+        time_between_hop: Duration::from_millis(175),
+        epoch_duration: Duration::from_secs(60_000),
+        fluff_probability: 0.0, // we want this test to always stem
+        graph: Graph::FourRegular,
+    };
+
+    let (broadcast_svc, mut broadcast_rx) = mock_broadcast_svc();
+    let (outbound_peer_svc, mut outbound_rx) = mock_discover_svc();
+
+    let mut router = DandelionRouterBuilder::default()
+        .with_config(config)
+        .with_broadcast_svc(broadcast_svc)
+        .with_outbound_peer_discover(outbound_peer_svc)
+        .build();
+
+    const FROM_PEER: usize = 20;
+
+    // The router will panic if it attempts to flush.
+    broadcast_rx.close();
+
+    for _ in 0..30 {
+        router
+            .ready()
+            .await
+            .unwrap()
+            .call(DandelionRouteReq {
+                tx: 0_usize,
+                state: TxState::Stem { from: FROM_PEER },
```

Comment:
`FROM_PEER` only used once?

Reply:
Btw, not saying this is bad - it could actually be more clear, just wondering if it was intentional.

Reply:
I'll add it back, yeah it was